### PR TITLE
More MKS UI cleanup, optimization

### LINKED ISSUE #19867: compilation error
Board: MKS robin nano v2.
After update https://github.com/MarlinFirmware/Marlin/commit/4533a197d59ffcaccc4d476f06df7754b11b9b0c (i think this) atom throws compilation error:
In file included from .pio\libdeps\mks_robin_nano35\lvgl/lvgl.h:41:0,
                 from Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:24,
                 from Marlin\src\lcd\extui\lib\mks_ui\draw_motor_settings.cpp:26:
.pio\libdeps\mks_robin_nano35\lvgl/src/lv_objx/lv_btn.h:120:12: note: candidate: lv_obj_t* lv_btn_create(lv_obj_t*, const lv_obj_t*)
 lv_obj_t * lv_btn_create(lv_obj_t * par, const lv_obj_t * copy);
            ^~~~~~~~~~~~~
.pio\libdeps\mks_robin_nano35\lvgl/src/lv_objx/lv_btn.h:120:12: note:   candidate expects 2 arguments, 8 provided
In file included from Marlin\src\lcd\extui\lib\mks_ui\draw_motor_settings.cpp:26:0:
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:189:11: note: candidate: lv_obj_t* lv_btn_create(lv_obj_t*, lv_event_cb_t, int, lv_style_t*)
 lv_obj_t* lv_btn_create(lv_obj_t *par, lv_event_cb_t cb, const int id, lv_style_t *style);
           ^~~~~~~~~~~~~
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:189:11: note:   candidate expects 4 arguments, 8 provided
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:192:11: note: candidate: lv_obj_t* lv_btn_create(lv_obj_t*, lv_event_cb_t, int)
 lv_obj_t* lv_btn_create(lv_obj_t *par, lv_event_cb_t cb, const int id);
           ^~~~~~~~~~~~~
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:192:11: note:   candidate expects 3 arguments, 8 provided
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:195:11: note: candidate: lv_obj_t* lv_btn_create(lv_obj_t*, lv_coord_t, lv_coord_t, lv_coord_t, lv_coord_
t, lv_event_cb_t, int)
 lv_obj_t* lv_btn_create(lv_obj_t *par, lv_coord_t x, lv_coord_t y, lv_coord_t w, lv_coord_t h, lv_event_cb_t cb, const int id);
           ^~~~~~~~~~~~~
Marlin\src\lcd\extui\lib\mks_ui\draw_ui.h:195:11: note:   candidate expects 7 arguments, 8 provided
*** [.pio\build\mks_robin_nano35\src\src\lcd\extui\lib\mks_ui\draw_motor_settings.cpp.o] Error 1

### PR DESCRIPTION
**Pull request by rhapsodyv**
### Description

Follow up for 19843 when TMC is enabled. Added a test for that too.

### Benefits

Fix #19867 

### Related Issues

#19867 
#19843

### CODE REVIEW

File: Marlin/src/lcd/extui/lib/mks_ui/draw_ready_print.cpp [hunk 1]
```diff
@@ -237,34 +237,9 @@ void lv_draw_ready_print(void) {
...
-    #if HAS_ROTARY_ENCODER
-      if (gCfgItems.encoder_enable == true) {
-        lv_group_add_obj(g, buttonTool);
-        lv_group_add_obj(g, buttonSet);
-        lv_group_add_obj(g, buttonPrint);
-      }
-    #endif
+    lv_big_button_create(scr, "F:/bmp_tool.bin", main_menu.tool, 20, 90, event_handler, ID_TOOL);
+    lv_big_button_create(scr, "F:/bmp_set.bin", main_menu.set, 180, 90, event_handler, ID_SET);
+    lv_big_button_create(scr, "F:/bmp_printing.bin", main_menu.print, 340, 90, event_handler, ID_PRINT);
```
Comment (rhapsodyv) [inline Marlin/src/lcd/extui/lib/mks_ui/draw_ready_print.cpp:221]: @thinkyhead this is what I meant about the "mks button". I created it: a function that create a full mks button, with label, and everything. So, more and more code can be removed.

**Comment by thinkyhead** [comment]
I'm pretty sure I made code that uses `toggle_stepping_mode` expect the new value, not the old value. If this is not the case, please change the code that uses `toggle_stepping_mode` instead of changing `toggle_stepping_mode`.

**Comment by thinkyhead** [comment]
Ah, I see you are adding your own. Well, please use this instead:
```cpp
inline bool toggle_stepping_mode() { set_stealthChop(!this->stored.stealthChop_enabled); return get_stealthChop(); }
```
Yes, it does return the actual hardware status, so if toggle is failing to change the hardware state, it will be obvious in the interface.

**Comment by rhapsodyv** [comment]
I created helpers to draw every type of "default" lvgl button and menu. I'm updating the code to use the new helpers, and removing the repeated code.
I think there're more up to 10 screens to update... this is endless...

**Comment by thinkyhead** [comment]
> I think there're more up to 10 screens to update... this is endless...

Welcome to my world! But the repetitive things that I can automate, I do tend to automate. For example, today I am adding the option to download and build one or more example configs to my `mftest` script as a basis for future CI on the Configurations repo. And, well, it would also produce BINs for all the configs, so those could be posted someplace too. But I digress……

File: Marlin/src/lcd/extui/lib/mks_ui/draw_acceleration_settings.cpp [hunk 2]
```diff
@@ -160,200 +160,59 @@ static void event_handler(lv_obj_t * obj, lv_event_t event) {
...
+    lv_screen_menu_item_1_edit(scr, machine_menu.RetractAcceleration, PARA_UI_POS_X, PARA_UI_POS_Y * 2, event_handler, ID_ACCE_RETRA, 1, public_buf_l);
 
-    (void)lv_label_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 4 + 10, machine_menu.X_Acceleration);
+    ZERO(public_buf_l);
+    sprintf_P(public_buf_l, PSTR("%.1f"), planner.settings.travel_acceleration);
+    lv_screen_menu_item_1_edit(scr, machine_menu.TravelAcceleration, PARA_UI_POS_X, PARA_UI_POS_Y * 3, event_handler, ID_ACCE_TRAVEL, 2, public_buf_l);
 
-    buttonXValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y * 4 + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_X);
-    labelXValue = lv_label_create_empty(buttonXValue);
+    ZERO(public_buf_l);
```
Comment (thinkyhead) [inline Marlin/src/lcd/extui/lib/mks_ui/draw_acceleration_settings.cpp]: I don't think these are needed in front of string population. The `sprintf_P` function is not likely to bail and return without writing to the buffer.

Comment (rhapsodyv) [reply Marlin/src/lcd/extui/lib/mks_ui/draw_acceleration_settings.cpp]: True.. I'm too much focused in moving and rewriting those menus that something pass. I will remove all those zero.

**Review by rhapsodyv — COMMENTED** [review]
> Ah, I see you are adding your own. Well, please use this instead:
> 
> ```c++
> inline bool toggle_stepping_mode() { set_stealthChop(!this->stored.stealthChop_enabled); return get_stealthChop(); }
> ```
> 
> Yes, it does return the actual hardware status, so if toggle is failing to change the hardware state, it will be obvious in the interface.

set_stealthChop don't exist yet..

File: Marlin/src/lcd/extui/lib/mks_ui/draw_ready_print.cpp (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/lcd/extui/lib/mks_ui/draw_ready_print.cpp:221]: I would have done more, but I had to sleep! I'm kind of surprised no one ever optimized this code before.

File: Marlin/src/lcd/extui/lib/mks_ui/draw_tmc_step_mode_settings.cpp [hunk 3]
```diff
@@ -260,7 +217,9 @@ void lv_draw_tmc_step_mode_settings(void) {
     line1 = lv_line_create(scr, NULL);
     lv_ex_line(line1, line_points[0]);
 
-    buttonYText = lv_btn_create(scr, NULL, PARA_UI_POS_X, PARA_UI_POS_Y * 2, PARA_UI_VALUE_SIZE_X, PARA_UI_SIZE_Y, event_handler, 0);
+    buttonYText = lv_btn_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 2, PARA_UI_VALUE_SIZE_X, PARA_UI_SIZE_Y, event_handler, 0);
```
Comment (thinkyhead) [inline Marlin/src/lcd/extui/lib/mks_ui/draw_tmc_step_mode_settings.cpp]: Oops! I must have had an imgbtn fart here…

**Comment by rhapsodyv** [comment]
it's never ends...

**Comment by thisiskeithb** [comment]
Here's another bug that came out of that MKS UI slim down: https://github.com/MarlinFirmware/Marlin/issues/19872

**Comment by rhapsodyv** [comment]
I think I finished. I have refactored all simple/standard/template screens (menus/editors/switch/big buttons).

Russian Lang worked for me.

I tested what I could. I think everything is ok... but as I changed almost all screens, I can be missing one.

**Pull request merged**